### PR TITLE
PVR API 5.1.0

### PR DESCRIPTION
**Pull request by ksooo**
Refer to xbmc/xbmc#9290 and xbmc/xbmc#9295 for details.

@janbar demuxer changes must be runtime tested before merge. Could you please do this and report back?

### CODE REVIEW

**Comment by janbar** [comment]
Yes Kai, sure. I will let you know as soon as possible.

**Comment by janbar** [comment]
What about reordering of streams ? Is it always required ?

**Comment by ksooo** [comment]
> What about reordering of streams ? Is it always required ?

I'm not an expert on this. I just 'copied' what @FernetMenta did for vdr.vnsi and he said my changes are good

**Comment by janbar** [comment]
Okay, so i will try like that ;) and let to see

**Comment by janbar** [comment]
If it need some change to run I will do a PR for you.

**Comment by ksooo** [comment]
> If it need some change to run I will do a PR for you.

 Cool. Thx.

**Comment by janbar** [comment]
Don't work. We played, we loose ;). I have to refactor it.

**Comment by janbar** [comment]
Kai, I think your changes are right, and the root cause is other. Is 9290 works for an other addon than vnsi ?

**Comment by janbar** [comment]
I reverted all and then keeping the header of the stream helper (xbmc_stream_utils.hpp) with tiny changes (just renaming PhysicalId and Identifier as required), the result is same: no work. I suspect some changes in specif of Kodi itself. Let me know if you succeeded to run an other addon handling demuxing. Thanks

**Comment by ksooo** [comment]
Sorry, don't know. The only other one is pvr.hts and I had not time to test it yet.

**Comment by ksooo** [comment]
> I reverted all and then keeping the header of the stream helper (xbmc_stream_utils.hpp) with tiny changes (just renaming PhysicalId and Identifier as required), the result is same: no work. I suspect some changes in specif of Kodi itself.

 This sounds not good. @FernetMenta anything we can do on the addon-side or is it kodi core problem?

**Comment by FernetMenta** [comment]
i am on a business trip and only have very limited internet connection. con't do much until Friday

**Comment by janbar** [comment]
I suspect the issue isn't new with this PR, but probably demuxing doesn't work for a moment. I haven't tested it for one month with lastest kripton. So i will rollback my kodi sources to find the devil commit causing this state.

**Comment by janbar** [comment]
For info, I rollbacked to api 5.0.0 (kodi commit a0f046c30768b0a0c40173faf6046c950f57e321), and it works

**Comment by janbar** [comment]
So the commit 03aa2441d3235880c939d92f77d7319fca4b5015 breaks pvr.mythtv.

**Comment by janbar** [comment]
@FernetMenta , any idea about ?

**Comment by FernetMenta** [comment]
I don't see that you set m_streams.stream[count].iPID
check out https://github.com/xbmc/xbmc/commit/0aa55897494fe6004b1dd9aa9a8a8579cb5bf6a3

**Comment by janbar** [comment]
Yes good point. The change here doesn't set the PID for new stream, like it was done by calling XbmcPvrStream::GetStreamData(pid, &new_stream).
I try to fix it.

**Comment by janbar** [comment]
I found others issues as properties are never copied when calling GetStreamProperties().
@FernetMenta, are streams have to be reordered as before, and placing main stream at index 0 of the array ?

**Comment by FernetMenta** [comment]
The index does not matter because streams are identified by their PIDs.

**Comment by janbar** [comment]
Thanks.
Until now it doesn't work. Always same error:
`ERROR: ReadPacket - Error demux packet doesn't belong to a valid stream`

I send logs at http://pastebin.com/4VRSKcHH

**Comment by janbar** [comment]
@FernetMenta , having a look in your change for vnsi, i see you set packet iStreamId with the PID now.

**Comment by janbar** [comment]
https://github.com/kodi-pvr/pvr.vdr.vnsi/commit/bb9a9b9d21668cb2e5ad3268dd6571511da3b927#diff-06ab38b3319cb40e878f211e20a5a26cR124

I try it now ;).

File: src/demux.cpp [hunk 1]
```diff
@@ -558,7 +573,18 @@ DemuxPacket* Demux::stream_pvr_data(TSDemux::STREAM_PKT* pkt)
...
+    for (int i = 0; i < m_streams.iStreamCount; i++)
+    {
+      if (m_streams.stream[i].iPID == (unsigned int)pkt->pid)
+      {
+        idx = i;
+        break;
+      }
+    }
+
+    dxp->iStreamId = idx;
```
Comment (FernetMenta) [inline src/demux.cpp]: what's that? iStreamId needs to be PID

**Comment by janbar** [comment]
It works now !

**Comment by FernetMenta** [comment]
great!

**Comment by janbar** [comment]
So I do some test with different stream types and then i will make a PR for @ksooo including our fixes.

Thanks @FernetMenta for your time.

**Comment by ksooo** [comment]
@janbar thx. no new pr needed. just point me to your commit and i will cherry-pick into this pr.

**Comment by janbar** [comment]
@ksooo , from my repo the commit is 500d0bf6b319114263faef39f188b0ffbd46a6af

**Comment by ksooo** [comment]
@janbar cherry-picked your commit

**Pull request merged**